python/cuml/cuml/model_selection/_split.py

```diff
@@ -419,7 +421,106 @@ def _process_df_objs(
         return X_train, X_test
 
 
-class StratifiedKFold:
+class _KFoldBase(ABC):
+    """Base class for k-fold split."""
+
+    def __init__(self, n_splits=5, shuffle=False, random_state=None):
+        if n_splits < 2 or not isinstance(n_splits, int):
+            raise ValueError(
+                f"n_splits {n_splits} is not a integer at least 2"
+            )
+
+        self.n_splits = n_splits
+        self.shuffle = shuffle
+        self.seed = random_state
+
+    @abstractmethod
+    def split(self, x, y):
+        """Generate indices to split data into training and test set.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data, where `n_samples` is the number of samples
+            and `n_features` is the number of features.
+
+        y : array-like of shape (n_samples,), default=None
+            The target variable for supervised learning problems.
+
+        Yields
+        ------
+        train_idx : CuPy ndarray
+            The training set indices for that split.
+
+        test_idx : CuPy ndarray
+            The testing set indices for that split.
+        """
+        raise NotImplementedError()
+
+
+class KFold(_KFoldBase):
+    """K-Folds cross-validator.
+
+    Provides train/test indices to split data in train/test sets. Split dataset into k
+    consecutive folds (without shuffling by default).
+
+    Each fold is then used once as a validation set while the k - 1 remaining folds form
+    the training set.
+
+    Parameters
+    ----------
+    n_splits :
+        Number of folds. Must be at least 2.
+
+    shuffle :
+        Whether to shuffle the samples before splitting.
+
+    random_state : int, CuPy RandomState or NumPy RandomState optional
+        If shuffle is true, seeds the generator. Unseeded by default.  Pass an int for
+        reproducible output across multiple function calls.
+
+    """
+
+    def __init__(self, n_splits=5, shuffle=False, random_state=None):
```

Comment:
We should be using keyword-only arguments after `n_splits`.

```suggestion
    def __init__(self, n_splits=5, * shuffle=False, random_state=None):
```

This is a general rule, but also matches the sklearn API.